x-pack/filebeat/docs/inputs/input-azure-blob-storage.asciidoc

```diff
@@ -224,10 +224,9 @@ This is a specific subfield of a container. It specifies the container name.
 [float]
 ==== `max_workers`
 
-This attribute defines the maximum number of workers (go routines / lightweight threads) are allocated in the worker pool (thread pool) for processing jobs 
-which read contents of file. More number of workers equals a greater amount of concurrency achieved. There is an upper cap of `5000` workers per container that 
-can be defined due to internal sdk constraints. This attribute can be specified both at the root level of the configuration as well at the container level. 
-The container level values will always take priority and override the root level values if both are specified.
+This attribute defines the maximum number of workers (goroutines / lightweight threads) are allocated in the worker pool (thread pool) for processing jobs which read the contents of files. This attribute can be specified both at the root level of the configuration and at the container level. Container level values override the root level values if both are specified. Larger number of workers do not necessarily improve of throughput, and this should be carefully tuned based on the number of files, the size of the files being processed and resources available. Increasing `max_workers` to very high values may cause resource utilization problems and can lead to a bottleneck in processing. Usually a maximum cap of `2000` workers is recommended. A very low `max_worker` count will drastically increase the number of network calls required to fetch the blobs, which can cause a bottleneck in processing.
+
+NOTE: The value of `max_workers` is tied to the `batch_size` currently to ensure even distribution of workloads across all goroutines. This ensures that the input is able to process the files in an efficient manner. This `batch_size` determines how many blobs will be fetched in one single call. The `max_workers` value should be set based on the number of files to be read, the resources available and the network speed. For example,`max_workers=3` would mean that every pagination request a total number of `3` blobs are fetched and distributed among `3 goroutines`, `max_workers=100` would mean `100` blobs are fetched in every pagination request and distributed among `100 goroutines`.
```

Comment:
```suggestion
The value of `max_workers` is tied to the `batch_size` currently to ensure even distribution of workloads across all workers. This ensures that the input is able to process the files in an efficient manner. This `batch_size` determines how many blobs will be fetched in one single call. The `max_workers` value should be set based on the number of files to be read, the resources available and the network speed. For example,`max_workers=3` would mean that every pagination request a total number of `3` blobs are fetched and distributed among `3 workers`, `max_workers=100` would mean `100` blobs are fetched in every pagination request and distributed among `100 workers`.
```

I don't understand what "tied' means here. I don't think that it means what it is intended to mean though. "tied" should mean that we do work to ensure that it is modified based on the value of `batch_size` which AFAIUI we do not.

I've replaced all instances of goroutines with workers since a goroutine is a technical detail that a user should not be interested in. They want to know (at most, likely) that there is concurrency and worker embodies that.

Reply:
I feel like we might change the implementation on batch size later down the road since there can be scenarios when having a 1:1 distribution might not be ideal. Also decoupling this would give the users more flexibility in controlling their desired batch size without messing with the worker count. What do you think @efd6 ?

Reply:
I don't yet know what the terms are here, so I can't answer that.

Reply:
updated this section